attest/eventlog.go

```diff
@@ -241,46 +249,88 @@ func extend(pcr, replay []byte, e rawEvent) ([]byte, Event, error) {
 			hash.Write(b)
 		}
 		hash.Write(digest)
-		return hash.Sum(nil), Event{e.index, e.typ, e.data, digest}, nil
+		return hash.Sum(nil), digest, nil
 	}
-	return nil, Event{}, fmt.Errorf("no event digest matches pcr length: %d", len(pcr))
+	return nil, nil, fmt.Errorf("no event digest matches pcr length: %d", len(pcr))
 }
 
-func replayEvents(rawEvents []rawEvent, pcrs []PCR) ([]Event, error) {
-	events := []Event{}
-	replay := map[int][]byte{}
-	pcrByIndex := map[int][]byte{}
-	for _, pcr := range pcrs {
-		pcrByIndex[pcr.Index] = pcr.Digest
-	}
+// replayPCR replays the event log for a specific PCR, using pcr and
+// event digests with the algorithm in pcr. An error is returned if the
+// replayed values do not match the final PCR digest, or any event tagged
+// with that PCR does not posess an event digest with the specified algorithm.
+func replayPCR(rawEvents []rawEvent, pcr PCR) ([]Event, error) {
```

Comment:
Since replayPCR's error is never reported, can the return value be `[]Event, bool`?